tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1201,19 +1217,36 @@ class CudnnRnnSequenceTensorDescriptor
                                    cudnnDataType_t data_type,
                                    TensorDescriptor handle)
       : parent_(parent),
-        seq_length_(seq_length),
+        max_seq_length_(seq_length),
         batch_size_(batch_size),
         data_size_(data_size),
         data_type_(data_type),
         handle_(std::move(handle)),
         handles_(seq_length, handle_.get()) {}
 
+  CudnnRnnSequenceTensorDescriptor(CUDAExecutor* parent, int seq_length,
+                                   int batch_size, int data_size,
+                                   int* seq_lens,
+                                   cudnnDataType_t data_type,
+                                   RNNDataDescriptor data_handle,
+                                   TensorDescriptor handle)
+      : parent_(parent),
+        max_seq_length_(seq_length),
+        batch_size_(batch_size),
+        data_size_(data_size),
+        seq_lens_(seq_lens),
+        data_type_(data_type),
+        rnn_data_handle_(std::move(data_handle)),
+        handle_(std::move(handle)),
+        handles_(seq_length, handle_.get()) {}
+
  public:
   CudnnRnnSequenceTensorDescriptor(CudnnRnnSequenceTensorDescriptor&&) =
       default;
 
   static port::StatusOr<CudnnRnnSequenceTensorDescriptor> Create(
       CUDAExecutor* parent, int seq_length, int batch_size, int data_size,
+      int* seq_lens,  // This is the host memory
```

Comment:
create a separate `Create` to for `seq_lengths` argument, rename `seq_length` above to `max_seq_length`.